lib/hammer_cli_foreman_puppet.rb

```diff
@@ -49,4 +49,10 @@ module HammerCLIForemanPuppet
     'HammerCLIForemanPuppet::SmartClassParameter',
     'hammer_cli_foreman_puppet/smart_class_parameter'
   )
+
+  # plugins extensions
+  begin
+    require 'hammer_cli_foreman_puppet/discovery'
+  rescue LoadError
```

Comment:
I guess it's better to use a general exception name, since e.g. for local tests it will raise `Resource 'discovered_hosts' does not exist in the API (NameError)`

Reply:
wouldn't rescue a general `Exception` be a bit risky? 
how about 
```ruby
EXCEPTIONS = [LoadError, NameError]

begin
  require 'hammer_cli_foreman_puppet/discovery'
rescue *EXCEPTIONS
end
```

Reply:
Well, I guess we could also include `ArgumentError` or even other ones, but not sure...

Why I suggest to rescue all of them is that if that plugin is broken for some reason we will still load puppet plugin. Since discovery is not a dependency it doesn't make much sense to fail to load if that plugin is somehow broken. It's up to that plugin to fix itself. The only problem I see is for developers of puppet plugin: one can not notice if they broke discovery plugin or their changes don't have any effect, so it should be dealt with carefuly.

And what do you mean by risky? Do you have ideas or more explicit concerns? Maybe I'm missing something.

Reply:
> And what do you mean by risky? 

I was thinking that some errors would be important to throw (maybe some errors related to the discovery's command extensions in hcf-puppet, but overall I agree,  It doesn't make sense to fail hcf-puppet from loading if discovery is broken. how about some indication? printing warnings to the console (maybe just in debug mode?)

Reply:
I still see that you rescue only one specific error, I'd vote for a general one, but with debug message in logs.

Reply:
> I still see that you rescue only one specific error, I'd vote for a general one, but with debug message in logs.

I think that makes sense. If anything does not work, you would run `hammer -d` and see what went wrong.